From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 18 Aug 2024 17:42:14 -0700
Subject: [1/2] meson: add detection logic for v4l2request support
Origin: https://github.com/mpv-player/mpv/pull/14690/commit/ca9271bb897133cc0dff50eaf73d1f0b86a8b2ab

We will probably adjust this to look for a specific libavutil version after
v4l2request support is merged upstream, but this check is fine for now.
---
Update patch for ``meson_options.txt`` to ``meson.options`` rename in upstream commit
c60d8523edec ("meson: rename meson_options.txt to meson.options")

 meson.build   | 9 +++++++++
 meson.options | 1 +
 2 files changed, 10 insertions(+)

diff --git a/meson.build b/meson.build
index 7572769e0e..46c69ea025 100644
--- a/meson.build
+++ b/meson.build
@@ -1411,6 +1411,14 @@ if features['ios-gl']
     sources += files('video/out/hwdec/hwdec_ios_gl.m')
 endif
 
+v4l2request = get_option('v4l2request').require(
+    cc.has_header_symbol('libavutil/hwcontext.h', 'AV_HWDEVICE_TYPE_V4L2REQUEST')
+)
+features += {'v4l2request': v4l2request.allowed()}
+if features['v4l2request']
+    sources += files('video/v4l2request.c')
+endif
+
 libva = dependency('libva', version: '>= 1.1.0', required: get_option('vaapi'))
 
 vaapi_drm = dependency('libva-drm', version: '>= 1.1.0', required:
@@ -1868,6 +1876,7 @@ summary({'d3d11': features['d3d11'],
          'libmpv': get_option('libmpv'),
          'lua': features['lua'],
          'opengl': features['gl'],
+         'v4l2request': features['v4l2request'],
          'vulkan': features['vulkan'],
          'wayland': features['wayland'],
          'x11': features['x11']},
diff --git a/meson.options b/meson.options
index dae0a333ef..3dba7a1451 100644
--- a/meson.options
+++ b/meson.options
@@ -99,6 +99,7 @@ option('d3d-hwaccel', type: 'feature', value: 'auto', description: 'D3D11VA hwac
 option('d3d9-hwaccel', type: 'feature', value: 'auto', description: 'DXVA2 hwaccel')
 option('gl-dxinterop-d3d9', type: 'feature', value: 'auto', description: 'OpenGL/DirectX DXVA2 hwaccel')
 option('ios-gl', type: 'feature', value: 'auto', description: 'iOS OpenGL ES interop support')
+option('v4l2request', type: 'feature', value: 'auto', description: 'V4L2 Request API hwaccel')
 option('videotoolbox-gl', type: 'feature', value: 'auto', description: 'Videotoolbox with OpenGL')
 option('videotoolbox-pl', type: 'feature', value: 'auto', description: 'Videotoolbox with libplacebo')
 
-- 
2.45.2

